adoc/chapters/programming_interface.adoc

```diff
@@ -17647,40 +17688,666 @@ member function template.
 
 
 [[swizzled-vec-class]]
-==== Swizzled [code]#vec# class
-
-The [code]#+__swizzled_vec__+# class must define an unspecified temporary which
-provides the entire interface of the SYCL [code]#vec# class template, including
-swizzled member functions, with the additions and alterations described below.
-The member functions of the [code]#+__swizzled_vec__+# class behave as though
-they operate on a [code]#vec# that is the result of the swizzle operation.
-
-  * The [code]#+__swizzled_vec__+# class template must be readable as an r-value
-    reference on the RHS of an expression.
-    In this case the swizzle operation is performed on the RHS of the expression
-    and then the result is applied to the LHS of the expression.
-  * The [code]#+__swizzled_vec__+# class template must be assignable as an
-    l-value reference on the LHS of an expression.
-    In this case the RHS of the expression is applied to the original SYCL
-    [code]#vec# which the [code]#+__swizzled_vec__+# represents via the swizzle
-    operation.
-    Note that a [code]#+__swizzled_vec__+# that is used in an l-value expression
-    may not contain any repeated element indexes.
-+
-For example: [code]#f4.xxxx() = fx.wzyx()# would not be valid.
-  * The [code]#+__swizzled_vec__+# class template must be convertible to an
-    instance of SYCL [code]#vec# with the type [code]#DataT# and number of
-    elements specified by the swizzle member function, if [code]#NumElements >
-    1#, and must be convertible to an instance of type [code]#DataT#, if
-    [code]#NumElements == 1#.
-  * The [code]#+__swizzled_vec__+# class template must be non-copyable,
-    non-moveable, non-user constructible and may not be bound to a l-value or
-    escape the expression it was constructed in.
-    For example [code]#auto x = f4.x()# would not be valid.
-  * The [code]#+__swizzled_vec__+# class template should return
-    [code]#+__swizzled_vec__&+# for each operator inherited from the [code]#vec#
-    class template interface which would return [code]#vec<DataT,
-    NumElements>&#.
+==== The swizzled vector classes
+
+The [code]#+__writeable_swizzle__+# and [code]#+__const_swizzle__+# classes are
+each views over a [code]#vec# object which captures the effects of a swizzle
+operation without actually performing that operation.
```

Comment:
There are a lot of occurrences of the phrase "the swizzle operation", and I don't think it will make the spec more clear if we change them all to be plural.

Here's my rationale for keep the singular form ... if you apply a swizzle operation to a swizzled vec object, the second swizzle operation is combined with the first swizzle operation, creating a new swizzle operation.  The new swizzled vec object represents this new swizzle operation, but it's still just one operation.